api/v1alpha1/clusterextension_types.go

```diff
@@ -43,6 +43,25 @@ const (
 	UpgradeConstraintPolicyIgnore UpgradeConstraintPolicy = "Ignore"
 )
 
+// Similar to NamespacedName, but with json
+type ClusterExtensionSecretRef struct {
+	// Name of the secret
+	Name string `json:"name"`
+	// Namespace of the secret
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Should we force this to be the install namespace?

Reply:
I actually disagree, the registry and it's secret are likely in a separate namespace from the installation namespace.